testsuite/oauth/smart.py

```diff
@@ -0,0 +1,105 @@
+""" The SMART oAuth strategy. """
+import requests
+
+
+ERROR_AUTH_REQUEST = """
+Authenticated POST to {url} returned {status_code}.
+{text}
+"""
+
+
+class SmartStrategy(object):
+    """ Implements the lib.oauth.Strategy interface. """
```

Comment:
Can you clarify whether this strategy is designed to work *only* with SMART's reference server, or whether you intend to generalize it for allscripts, Cerner, Epic etc? If it's to be generalized, the only ways I really know are

##### 1. as a series of form-fill expectations and requests, to be used in a `requests.session`, like:

```python
[
  ["load", authorize_url],
  ["submitForm", ["form.someSelector", {
    "username": "argonaut",
    "password": "argonaut"
  }]
]
```

##### 2. A series of clicks and keystrokes, to be used via webdriver, like:

```python
 [
    {"click": "#txtUsername"},
    {"keys": "ARGONAUT"},
    {"click": "#txtPassword"},
    {"keys": "ARGONAUT"},
    {"click": "#cmdLogin"}
]
```